packages/typespec-test/test/batch_modular/generated/typespec-ts/src/BatchService.ts

```diff
@@ -0,0 +1,815 @@
+  listPools,
+  PoolDeletePoolOptions,
+  deletePool,
+  PoolExistsOptions,
+  exists,
+  PoolGetPoolOptions,
+  getPool,
+  PoolPatchPoolOptions,
+  patchPool,
+  PoolDisableAutoScaleOptions,
+  disableAutoScale,
+  PoolEnableAutoScaleOptions,
+  enableAutoScale,
+  PoolEvaluateAutoScaleOptions,
+  evaluateAutoScale,
+  PoolResizeOptions,
+  resize,
+  PoolStopResizeOptions,
+  stopResize,
+  PoolUpdatePropertiesOptions,
+  updateProperties,
+  PoolRemoveNodesOptions,
+  removeNodes,
+} from "./api/Pool.js";
+import {
+  TaskAddTaskOptions,
+  addTask,
+  TaskListTasksOptions,
+  listTasks,
+  TaskAddTaskCollectionOptions,
+  addTaskCollection,
+  TaskDeleteTaskCollectionOptions,
+  deleteTaskCollection,
+  TaskGetTaskCollectionOptions,
+  getTaskCollection,
+  TaskUpdateTaskCollectionOptions,
+  updateTaskCollection,
+  TaskListSubtasksOptions,
+  listSubtasks,
+  TaskTerminateTaskCollectionOptions,
+  terminateTaskCollection,
+  TaskReactivateTaskCollectionOptions,
+  reactivateTaskCollection,
+} from "./api/Task.js";
+import { ClientOptions } from "./common/interfaces.js";
+import { BatchServiceContext } from "./rest/clientDefinitions.js";
+import {
+  PoolStatistics,
+  BatchPool,
+  AutoScaleRun,
+  JobStatistics,
+  BatchJob,
+  Certificate,
+  BatchJobSchedule,
+  BatchTask,
+} from "./rest/models.js";
+
+export class BatchService {
+  private _client: BatchServiceContext;
+
+  /** A client for issuing REST requests to the Azure Batch service. */
+  constructor(
+    endpoint: string,
+    credential: TokenCredential,
+    options: ClientOptions = {}
+  ) {
+    this._client = createBatchService(endpoint, credential, options);
+  }
+
+  applications = {
+    listApplications: (
+      options: ApplicationsListApplicationsOptions = { requestOptions: {} }
+    ): Promise<ApplicationListResult> => {
+      return listApplications(this._client, options);
+    },
+    get: (
+      applicationId: string,
+      options: ApplicationsGetOptions = { requestOptions: {} }
+    ): Promise<Application> => {
+      return get(this._client, applicationId, options);
+    },
+  };
+  pool = {
+    listUsageMetrics: (
+      options: PoolListUsageMetricsOptions = { requestOptions: {} }
+    ): Promise<CustomPagePoolUsageMetrics> => {
+      return listUsageMetrics(this._client, options);
+    },
+    getAllPoolLifetimeStatistics: (
+      options: PoolGetAllPoolLifetimeStatisticsOptions = { requestOptions: {} }
+    ): Promise<PoolStatistics> => {
+      return getAllPoolLifetimeStatistics(this._client, options);
+    },
+    addPool: (
+      options: PoolAddPoolOptions = { requestOptions: {} }
+    ): Promise<void> => {
+      return addPool(this._client, options);
+    },
+    listPools: (
+      options: PoolListPoolsOptions = { requestOptions: {} }
+    ): Promise<BatchPoolListResult> => {
+      return listPools(this._client, options);
+    },
+    deletePool: (
+      poolId: string,
+      options: PoolDeletePoolOptions = { requestOptions: {} }
```

Comment:
Why not below statement?

```suggestion
      options?: PoolDeletePoolOptions
```

Reply:
`{ requestOptions: {} }`  is creating a default options object with an empty request options object as a porperty. This saves us to have to nullcheck and create an object when we need to add properties to request options. Otherwise in the body we'll ahve to do:

```ts

   if(!options) {
      options = {requestOptions: {}};
   }

   if (!options.requestOptions) {
      options.requestOptions = {};
   }

   options.requestOptions.foo = "foo";

```

vs


```ts
   options.requestOptions.foo = "foo";
```